### DISCUSSION TITLE
problen with adding tunisian number to github

### DISCUSSION THREAD

**Discussion by dananothername** [community/community#69618] [New to GitHub] [answered]
as i said i can't find  my country in the letter t section
my country phone code is +216

**Comment by queenofcorgis** [comment]
If your country is not on [the list](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/countries-where-sms-authentication-is-supported#supported-countries-for-sms-authentication), then we aren't currently able to reliably deliver text messages to your country. We track delivery rates across all countries, including ones not in the list, and review it to understand when we have to remove countries or can add them back. We do not add countries with low deliverability rates, because it leads to users being locked out of their accounts if they can’t receive the SMS for 2FA.

**Reply by dananothername** [reply]
so i have to wait?

**Reply by DNin01** [reply, marked as answer]
Either you'll have to wait for your country to be added as an option or you'll have to configure an authenticator (TOTP) app instead. TOTP apps are actually preferred by GitHub as well as many individuals for privacy and security reasons.

**Reply by dananothername** [reply]
thanks it worked 
now that app keep running and generating 6 numbers
is that ok

**Reply by DNin01** [reply]
> thanks it worked 
> 
> now that app keep running and generating 6 numbers
> 
> is that ok
> 
> 

Yep! That's how authenticator apps work - they scan a QR code containing a secret key, which is then used to generate random numbers (technically cryptographically secure pseudorandom numbers). The number changes overtime, and by checking that the 6-digit code you entered when authenticating is what it should be in the authenticator app at the moment, a service can verify that the user is in possession of the device used to scan the QR code.

**Reply by queenofcorgis** [reply]
@dananothername happy to see you've got TOTP set up ✅ and thanks to @DNin01 for the detailed explanation!
